modules/derivation/shared/src/main/scala/io/circe/derivation/package.scala

```diff
@@ -7,10 +7,23 @@ package object derivation {
   final def deriveEncoder[A]: Encoder.AsObject[A] = macro DerivationMacros.materializeEncoder[A]
   final def deriveCodec[A]: Codec.AsObject[A] = macro DerivationMacros.materializeCodec[A]
 
-  final def deriveDecoder[A](nameTransformation: String => String): Decoder[A] =
+  final def deriveDecoder[A](
+    nameTransformation: String => String,
+    useDefaults: Boolean,
```

Comment:
What do you think about putting defaults on `useDefaults` and `discriminator` to support the previous usage transparently.

Reply:
I was thinking it, but I discovered that you cannot use default values mixed with macros.

Reply:
Oh, right. This is fine with me, then.